cli/main.py

```diff
@@ -1018,14 +1009,31 @@ def __make_ursula():
     config.operating_mode = "federated" if ursula_config.federated_only else "decentralized"
     click.secho("Running in {} mode".format(config.operating_mode), fg='blue')
 
-    if trusted_teacher_uri:
-        address, rest_uri = trusted_teacher_uri.split("@")
-        host, port = rest_uri.split(":")
-        seed_metadata = [SeednodeMetadata(address, host, port)]  # TODO: Ensure seed injection works
-    else:
-        seed_metadata = ()
+    #
+    # Seed
+    #
+    seed_metadata, teacher_nodes = list(), list()
+    if teacher_uri:
+
+        if all(('@' in teacher_uri, ':' in teacher_uri)):
```

Comment:
This seems like odd logic to me.  If they don't specify a port (but do specify a checksum address) then we shove them into `from_seed_and_stake_info`?

Reply:
Validation regex?

Reply:
Yeah - this needs a bit of work, fixing #522 will start to open this logic a bit